### PR TITLE
[Elastic Agent] Fix merging of fleet.yml. Add --staging to enroll cmd.

### LINKED ISSUE #19961: [Ingest Manager] 7.9 BC1 Agent is not able to download a corresponding 7.9 Endpoint-Security artifact from the BC location
7.9 BC1 Agent is not able to download a corresponding 7.9 Endpoint-Security artifact from the BC location

For confirmed bugs, please report:
- Version: 7.9 BC1 (and latest 7.9 Stack code in cloud staging deploy)
- Operating System: win 7 x64 for Agent.
- Steps to Reproduce: 
1) deploy the latest 7.9 stack to cloud, i"m using SNAPSHOT available as of 5PM 7/15
2) setup Ingest Manager / Fleet and copy the enrollment command for the default config
3) download the BC agent from this location to the host: https://staging.elastic.co/7.9.0-33ee764c/summary-7.9.0.html
4) unarchive and enroll and run agent.  all is well, so far.
5) create a config that has 'Endpoint' integration,
6) then change your running Agent to use this new config.
7) in Agent logs this is where I see the attempt to download Endpoint-security but it fails. 

`Application: endpoint-security--7.9.0[6c1f576d-ee62-4892-826d-f82ba2fe8371]: State changed to FAILED: 2 errors occurred: * package 'data\downloads\endpoint-security-7.9.0-windows-x86_64.zip' not found: open data\downloads\endpoint-security-7.9.0-windows-x86_64.zip: The system cannot find the file specified. * call to 'https://artifacts.elastic.co/downloads/endpoint-dev/endpoint-security-7.9.0-windows-x86_64.zip' returned unsuccessful status code: 404: /go/src/github.com/elastic/beats/x-pack/elastic-agent/pkg/artifact/download/http/downloader.go[142]: unknown error
`

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Fixes an issue where the `elastic-agent.yml` and `fleet.yml` configurations where being merged, but the merged config was not being used in the `Unpack` call.

Adds a `--staging f32a4a89` option to `./elastic-agent enroll`. This allows users that are testing staging builds to be able to pull artifacts from that staging build.

## Why is it important?

So `elastic-agent.yml` and `fleet.yml` are merged and used correctly. To simplify testing of staging builds.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

```
 ./elastic_agent enroll --insecure http://localhost:5601 Q1JfZVhITUJsUURwelBhQVB2WVQ6NmlFSVNwZVVSU0tlcVI5dXZNNUtFZw --staging f32a4a89
```

## Related issues

- Closes #19961

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/application/managed_mode.go [hunk 1]
```diff
@@ -81,9 +81,15 @@ func newManaged(
...
+	err = rawConfig.Merge(config)
+	if err != nil {
+		return nil, errors.New(err,
+			fmt.Sprintf("fail to merge configuration with %s for the elastic-agent", path),
+			errors.TypeConfig,
+			errors.M(errors.MetaKeyPath, path))
+	}
 
-	cfg, err := configuration.NewFromConfig(config)
+	cfg, err := configuration.NewFromConfig(rawConfig)
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/application/managed_mode.go:92]: this is a good catch

File: x-pack/elastic-agent/pkg/agent/cmd/enroll.go [hunk 2]
```diff
@@ -67,6 +68,13 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags, args
 			errors.M(errors.MetaKeyPath, pathConfigFile))
 	}
 
+	staging, _ := cmd.Flags().GetString("staging")
+	if staging != "" {
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/cmd/enroll.go:72]: could be a single line if

**Comment by EricDavisX** [comment]
@blakerouse this is great!  can you elaborate for the sake of over communicating on this:
 ./elastic_agent enroll --insecure http://localhost:5601 Q1JfZVhITUJsUURwelBhQVB2WVQ6NmlFSVNwZVVSU0tlcVI5dXZNNUtFZw --staging f32a4a89

where does the f32a4a89 come from... I assume this is a hash of the build artifacts, like this:
pulling '33ee764c' from https://staging.elastic.co/7.9.0-33ee764c/summary-7.9.0.html
 - which is the HTML page that comes from the 7.9 main tracking issue: https://github.com/elastic/dev/issues/1442

**Comment by blakerouse** [comment]
@EricDavisX Correct, it comes from the HTML page of the build.

**Comment by EricDavisX** [comment]
thanks.  this is so great!

**Comment by EricDavisX** [comment]
Also, does this PR fix the problem of Linux Agent not ‘finding’ Endpoint even when its on disk (manually copied there with the SHA 512 file)? Not sure if we need a new bug or not.
centos7-tar-gz-agent-endpoint-integration-problem

linux-file-for-endpoint-exist-but-not-found

**Comment by EricDavisX** [comment]
FYI - we found the bug on my side and it was a personal script problem, not in the code - yay.  we're good to go!

**Comment by EricDavisX** [comment]
note - this works great, tested with BC3 - we're in business!

**Pull request merged**